File: leetcode/137_single_number_ii.py
from typing import List


class Solution:

    def singleNumber(self, nums: List[int]) -> int:
        ones, twos, n = 0, -1, len(nums)
        for i in range(n):
            ones = ones ^ nums[i] & twos
            twos = twos ^ nums[i] & ones
        return ones

File: leetcode/java/137SingleNumberII.java
import java.util.HashMap;
import java.util.Map;

class Solution {
  public int singleNumber(int[] nums) {
    Map<Integer, Integer> numsCount = new HashMap<>();
    for (int num : nums) {
      numsCount.put(num, numsCount.getOrDefault(num, 0) + 1);
    }

    for (int num : numsCount.keySet()) {
      if (numsCount.get(num) == 1) {
        return num;
      }
    }
    return -1;
  }
}
